contributors/design-proposals/statefulset-update.md

```diff
@@ -0,0 +1,850 @@
+the tenant application has no way to determine if it is being terminated due to 
+a scale down operation or due to an update. 
+Consider a BASE distributed storage application like Cassandra, where 2 TiB of 
+persistent data is not atypical, and the data distribution is not identical on 
+every server. We want to enable two distinct behaviors based on reason for 
+termination.
+
+- If the termination is due to scale down, during the configured termination 
+grace period, the entry point of the Pod should cause the application to drain 
+its client connections, replicate its persisted data (so that the cluster is not 
+left under replicated) and decommission the application to remove it from the 
+cluster.
+- If the termination is due to a temporary capacity loss (e.g. an update or an 
+image upgrade), the application should drain all of its client connections, 
+flush any in memory data structures to the file system, and synchronize the 
+file system with storage media. It should not redistribute its data.
+
+If the application implements the strategy of always redistributing its data, 
+we unnecessarily decrease recovery time during an update and incur the 
+additional network and storage cost of two full data redistributions for every 
+updated node.
+It should be noted that this is already an issue for Node cordon and Pod eviction 
+(due to drain or taints), and applications can use the same mitigation as they 
+would for these events for StatefulSet update.
+
+### VolumeTemplatesSpec Updates
+While this proposal does not address 
+[VolumeTemplateSpec updates](https://github.com/kubernetes/kubernetes/issues/41015), 
+this would be a valuable feature for production users of storage systems that use
+intermittent compaction as a form of garbage collection. Application that use 
+log structured merge trees with size tiered compaction (e.g Cassandra) or append 
+only B(+/*) Trees (e.g Couchbase) can temporarily double their storage usage when 
+compacting their on disk storage. If there is insufficient space for compaction 
+to progress, these applications will either fail or degrade  until 
+additional capacity is added. While, if the user is using AWS EBS or GCE PD, 
+there are valid manual workarounds to expand the size of a PD, it would be 
+useful to automate the resize via updates to the StatefulSet's 
+VolumeClaimsTemplate.
+
+### In Place Updates
+Currently configuration, images, and resource request/limits updates are all 
+performed destructively. Without a [termination reason](https://github.com/kubernetes/kubernetes/issues/1462)
+implementation, there is little value to implementing in place image updates, 
+and configuration and resource request/limit updates are not possible.
+When [termination reason](#https://github.com/kubernetes/kubernetes/issues/1462) is implemented we may modify 
+the behavior of StatefulSet update to only update, rather than delete and 
```

Comment:
There are other reasons you may want new pods - this sounds like something that would be a strategy option (and applies to daemonsets as well).